pandas/core/indexes/multi.py

```diff
@@ -3644,6 +3644,10 @@ def _union(self, other, sort) -> MultiIndex:
             # This is only necessary if both sides have nans or one has dups,
             # fast_unique_multiple is faster
             result = super()._union(other, sort)
+
+            if isinstance(result, MultiIndex):
```

Comment:
Could `self._is_multi` be used?

Reply:
No, we have to check result, but result could also be an array, so have to be explicit here